launch/pipeline/service.py

```diff
@@ -1,18 +1,34 @@
 import inspect
+import uuid
 from typing import Any, Callable, Dict, List, Optional
 
 from launch.pipeline.deployment import Deployment
 from launch.pipeline.runtime import Runtime
 
+MAX_SERVICE_NAME_SIZE = 50
+
 
 class ServiceDescription:
     """
     Base ServiceDescription.
     """
 
+    service_name: str
+
+    def __init__(self):
+        self.make_request: Optional[Callable] = None
+
+    def set_make_request(
```

Comment:
This class should extend `abc.ABC`. And it's good that these methods raise `NotImplementedError`. They also should be decorated with `@abstractmethod` from `abc`.